### PR TITLE
feat: [IEG-3134] [FIMS] Pass the Mixpanel ID to enabled FIMS destinations

### PR DESCRIPTION
**Pull request by invyctus92**
## Short description
This pull request introduces a mechanism to selectively enrich FIMS redirect URLs with the Mixpanel device ID, but only for URLs explicitly allowed by remote configuration. The enrichment is controlled by feature flags and robustly handles error scenarios. The changes include new selectors, utility functions, saga logic, and comprehensive tests to ensure correct behavior.

This PR depends on https://github.com/pagopa/io-dev-api-server/pull/617

## List of changes proposed in this pull request
**FIMS Redirect URL Enrichment:**

* Added the `enrichFimsDestinationUrl` utility function, which appends the Mixpanel device ID to allowed redirect URLs, based on a configurable allowlist. If the URL is not allowed or is invalid, it is returned unchanged.
* Introduced the `enrichFimsRedirectUrl` saga, which uses the allowlist and Mixpanel enablement flag to determine if enrichment should occur, and safely handles device ID retrieval failures.
* Updated the FIMS authorization saga to use the enriched redirect URL when opening the authentication session.

**Configuration and State Management:**

* Added the `fimsTrackingEnrichedUrlsSelector` to retrieve the allowlist from remote configuration, with tests to verify correct selector behavior. [[1]](diffhunk://#diff-bbcaa878d6ddd74dcce6ecdc79cdc69a9b374700bfa387a58e9553a59ead7f4dR223-R232) [[2]](diffhunk://#diff-c334d60ada2ac4436f621e9955fb4059dffe762285635b8ed3c0309929a6ee8cR342-R360)

**Testing Enhancements:**

* Added unit tests for `enrichFimsDestinationUrl` and `enrichFimsRedirectUrl`, covering all relevant scenarios (Mixpanel enabled/disabled, device ID retrieval failure, allowed/disallowed URLs). [[1]](diffhunk://#diff-a32c47cbb3037b97820e15cb4ee3a0620c07f10d4d0f0fa2b18f58d5bdb4b27dR66-R96) [[2]](diffhunk://#diff-3001b79a0db6924c3ab679508a9eaa178201a1846eeff04ed8348cde3640069eR275-R321)

**Supporting Changes:**

* Updated the API models generation script to use the latest `io-services-metadata` branch for enriched URLs support.

These changes ensure that device tracking for FIMS flows is only performed for explicitly allowed URLs and only when Mixpanel tracking is enabled, improving privacy and configurability.

## How to test
all checks must pass

### CODE REVIEW

File: apps/main-app/scripts/generate-api-models.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 IO_BACKEND_VERSION=v20.0.0
 # need to change after merge on io-services-metadata
-IO_SERVICES_METADATA_VERSION=1.1.2
+IO_SERVICES_METADATA_VERSION=IEG-3133-FIMS-tracking-Enriched-Urls
```
Comment (Vangaorth) [inline apps/main-app/scripts/generate-api-models.sh]: Let's merge the PR on the CDN so we can properly reference the right version

File: apps/main-app/ts/store/reducers/backendStatus/remoteConfig.ts [hunk 2]
```diff
@@ -220,6 +220,16 @@ export const fimsServiceConfiguration = createSelector(
     )
 );
 
+export const fimsTrackingEnrichedUrlsSelector = createSelector(
```
Comment (Vangaorth) [inline apps/main-app/ts/store/reducers/backendStatus/remoteConfig.ts:223]: No more fp-ts code in new developments. This selector should be written to only use `O.toUndefined` in order to extract `value | undefined` from `remoteConfig`

File: apps/main-app/ts/store/reducers/backendStatus/__tests__/remoteConfig.test.ts [hunk 3]
```diff
@@ -338,6 +339,25 @@ describe("remoteConfig", () => {
...
+          fims: { trackingEnrichedUrls }
+        })
+      } as GlobalState;
+
+      expect(fimsTrackingEnrichedUrlsSelector(state)).toEqual(
+        trackingEnrichedUrls
+      );
+    });
+
+    it("should return an empty allowlist when it is not configured", () => {
```
Comment (Vangaorth) [inline apps/main-app/ts/store/reducers/backendStatus/__tests__/remoteConfig.test.ts:356]: We should also have a test that checks that an empty array is returned if the state is as follows:
`````
{ remoteConfig: O.some({ fims: {}}) }
`````

File: apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts [hunk 4]
```diff
@@ -45,6 +49,28 @@ import {
 
 // note: IAB => InAppBrowser
 
+export function* enrichFimsRedirectUrl(
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts]: What about moving this function at the end of the file, in order to have a better file's flow readability?

File: apps/main-app/ts/features/fims/singleSignOn/utils/index.ts [hunk 5]
```diff
@@ -77,3 +77,26 @@ export const removeFIMSPrefixFromUrl = (fimsUrlWithProtocol: string) => {
...
+ */
+export const enrichFimsDestinationUrl = (
+  destinationUrl: string,
+  trackingEnrichedUrls: ReadonlyArray<string>,
+  mixpanelDeviceId: string
+): string => {
+  try {
+    const parsedDestinationUrl = new URL(destinationUrl);
+    const destinationFullPath = `${parsedDestinationUrl.origin}${parsedDestinationUrl.pathname}`;
+    if (!trackingEnrichedUrls.includes(destinationFullPath)) {
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/utils/index.ts]: This comparison does not take into account case-sensitive urls. What do you think about writing a more robust solution like the one that follows (AI generated, so feel free to re-generate it or elaborate it a bit more through another AI):
`````
const normalizeUrl = (rawUrl: string): string => {
  try {
    const url = new URL(rawUrl);
    url.search = "";
    url.hash = "";
    return url.toString().toLowerCase().replace(/\/$/, "");
  } catch {
    return rawUrl.trim().toLowerCase().replace(/\/$/, "");
  }
};

export const enrichFimsDestinationUrl = (
  destinationUrl: string,
  trackingEnrichedUrls: ReadonlyArray<string>,
  mixpanelDeviceId: string
): string => {
  if (!destinationUrl || !mixpanelDeviceId) {
    return destinationUrl;
  }

  try {
    const parsedDestinationUrl = new URL(destinationUrl);
    const normalizedDestination = normalizeUrl(destinationUrl);

    const isMatch = trackingEnrichedUrls.some(
      (targetUrl) => normalizeUrl(targetUrl) === normalizedDestination
    );

    if (!isMatch) {
      return destinationUrl;
    }

    parsedDestinationUrl.searchParams.set("device", mixpanelDeviceId);
    return parsedDestinationUrl.toString();
  } catch {
    return destinationUrl;
  }
};
`````

File: apps/main-app/ts/features/fims/singleSignOn/utils/index.ts [hunk 6]
```diff
@@ -77,3 +77,26 @@ export const removeFIMSPrefixFromUrl = (fimsUrlWithProtocol: string) => {
...
+ * allowed by remote configuration. Query parameters and fragments do not take
+ * part in the allowlist comparison.
+ */
+export const enrichFimsDestinationUrl = (
+  destinationUrl: string,
+  trackingEnrichedUrls: ReadonlyArray<string>,
+  mixpanelDeviceId: string
+): string => {
+  try {
+    const parsedDestinationUrl = new URL(destinationUrl);
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/utils/index.ts]: Be aware that we are using 
``````
import {
  URL as PolyfillURL,
  URLSearchParams as PolyfillURLSearchParams
} from "react-native-url-polyfill";
``````
due to the URL apis having troubles on older react native versions. I suggest that we keep using the polyfilled version here and maybe later in the future switch back to the react-native one (in one shot, to test it once)

File: apps/main-app/ts/features/fims/singleSignOn/utils/index.ts [hunk 7]
```diff
@@ -77,3 +77,26 @@ export const removeFIMSPrefixFromUrl = (fimsUrlWithProtocol: string) => {
...
+    const destinationFullPath = `${parsedDestinationUrl.origin}${parsedDestinationUrl.pathname}`;
+    if (!trackingEnrichedUrls.includes(destinationFullPath)) {
+      return destinationUrl;
+    }
+    parsedDestinationUrl.searchParams.set("device", mixpanelDeviceId);
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/utils/index.ts]: If it is the mixpanelId, why do we call it `device` and not `mixpanelId`?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: apps/main-app/ts/features/fims/singleSignOn/utils/index.ts (hunk 7 above)
Comment (LeleDallas) [reply apps/main-app/ts/features/fims/singleSignOn/utils/index.ts]: After an internal discussion, we agree on that👍

**Comment by invyctus92** [comment]
@Vangaorth, Thanks for the comments; I'll fix it.

File: apps/main-app/ts/features/fims/singleSignOn/utils/__test__/index.test.ts [hunk 8]
```diff
@@ -59,3 +63,52 @@ describe("isIoFIMSLink", () => {
     });
   });
 });
+
+describe("enrichFimsDestinationUrl", () => {
+  const allowedUrl = "https://rp.example.it/fims/landing";
+  const deviceId = "mixpanel-device-id";
+
+  it.each([
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/utils/__test__/index.test.ts:71]: Should we also check for:
- no match for empty `trackingEnrichedUrls`
- a match on `trackingEnrichedUrls` with one item
- a match on `trackingEnrichedUrls` with two items, where the first does not match while the second does
?

File: apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts [hunk 9]
```diff
@@ -420,3 +428,25 @@ const inAppBrowserErrorToHumanReadable = (error: unknown) => {
...
+  redirectUrl: string
+): Generator<ReduxSagaEffect, string, any> {
+  const trackingEnrichedUrls = yield* select(fimsTrackingEnrichedUrlsSelector);
+  const mixpanelEnabled = yield* select(isMixpanelEnabled);
+  if (!mixpanelEnabled || trackingEnrichedUrls.length === 0) {
+    return redirectUrl;
+  }
+
+  try {
```
Comment (Vangaorth) [inline apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts]: Do we really need this try-catch?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts (hunk 9 above)
Comment (invyctus92) [reply apps/main-app/ts/features/fims/singleSignOn/saga/handleFimsAuthorizationOrImplicitCodeFlow.ts]: I'm removing the try-catch block; it's not needed.

**Review by Vangaorth — APPROVED** [review]
LGTM

**Review by LeleDallas — APPROVED** [review]
